src/ui/mainwindow.cpp

```diff
@@ -1672,15 +1670,20 @@ void MainWindow::on_documentLoaded(EditorTabWidget *tabWidget, int tab, bool was
 
         updateRecentDocsInMenu();
     }
+}
 
-    if (!wasAlreadyOpened) {
-        if (m_settings.General.getWarnForDifferentIndentation()) {
-            checkIndentationMode(editor);
+// TODO: Make editor aware if a file is already loaded so
+// indentation banner only appears for new files.
+void MainWindow::on_fileLoaded(bool wasAlreadyOpened)
```

Comment:
Unfortunately yes.... we're currently pretty heavily tied into how DocEngine functions... any fix I can come up with "quickly" will be hacky at best... so its best to leave that alone for another PR.